contracts/Libraries/Tupler.sol

```diff
@@ -0,0 +1,38 @@
+pragma solidity ^0.8.4;
+
+contract LibTupler {
+    function tupleSlicer(
```

Comment:
I think we can provide a more intuitive interface than this. Something like:
```
function extractElement(bytes memory tuple, uint index)
function extractDynamicElement(bytes memory tuple, uint index, uint nextIndex)
```

`extractElement` just returns the `index`th word of the tuple.
`extractDynamicElement` reads two offsets from words `index` and `nextIndex`, and returns that segment of the data. If `nextIndex` is zero, it returns the slice from `index` to the end of the string.